### PR TITLE
json-parser: parse 64 bit integer numbers

### LINKED ISSUE #3367: json parsing of int64 integers is wrong
# syslog-ng
## Version of syslog-ng
Latest OSE and PE releases

## Platform
Irrelevant

# Issue
## Failure
If we try to parse an int64 integer with a value larger than INT32_MAX with json-parser, then the result will be a 32-bit integer value set to INT32_MAX.

## Steps to reproduce
See the configuration

## Configuration
```
source s_local {
	example-msg-generator(template('{ "date": 1595441285858 }'));
};destination d_local {
	file("/dev/stdout" template("$(format-json date=int64(${.json.date}))\n"));
};parser p_json {
	json-parser(prefix('.json.'));
};log {
	source(s_local);
	parser(p_json);
	destination(d_local);
};
```

## Input and output logs (if possible)
```
[2020-07-30T08:59:20.499218] json-parser message processing started; input='{ "date": 1595441285858 }', prefix='.json.', marker='(null)', msg='0x55d8d3acc7a0'
[2020-07-30T08:59:20.499352] Setting value; name='.json.date', value='2147483647', msg='0x55d8d3acc7a0'
```

### PR DESCRIPTION
**Pull request by Kokan**
Fixes #3367

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by furiel — APPROVED** [review]

File: modules/json/json-parser.c [hunk 1]
```diff
@@ -109,8 +109,8 @@ json_parser_process_single(struct json_object *jso,
       break;
     case json_type_int:
       parsed = TRUE;
-      g_string_printf(value, "%i",
-                      json_object_get_int(jso));
+      g_string_printf(value, "%ld",
```
Comment (MrAnno) [inline modules/json/json-parser.c:112]: I think we should use `PRId64` or `G_GINT64_FORMAT`.

Comment (Kokan) [reply modules/json/json-parser.c]: `PRId64` it is.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by MrAnno — APPROVED** [review]
Thanks!

**Comment by Kokan** [comment]
Please note that the only failing travis job actually not an issue. I acidently pushed the same branch to syslog-ng project as well, but removed soon after. But as such that travis job could not find the referenced commit thus failed. The same travis job passed on the current fork.

**Review by szemere — APPROVED** [review]

**Pull request merged**